File: C_Prepend_and_Append.cpp
#include<bits/stdc++.h>
using namespace std;
void solve(){
    int n;
    cin>>n;
    string s;
    cin>>s;
    int l=0,r=n-1,ans=n;
    while(s[l]!=s[r] && ans>0){
        l++; r--; ans-=2;
    }
    cout<<ans<<endl;
}
int main(){
    int t;
    cin>>t;
    for(int i=0;i<t;i++){
        solve();
    }
}

File: C_Prepend_and_Append.py
def solve():
    n = int(input())
    s = input().strip()
    l, r = 0, n - 1
    ans = n
    while l <= r and s[l] != s[r] and ans > 0:
        l += 1
        r -= 1
        ans -= 2
    print(ans)

t = int(input())
for _ in range(t):
    solve()